mnt/reform2-nitrogen8m/uboot/default.nix

```diff
@@ -0,0 +1,16 @@
+{ buildUBoot, fetchgit }:
+
+buildUBoot rec {
+  pname = "uboot-reform2-imx8mq";
```

Comment:
Please move the packages to nixpkgs, we cannot test/build in nixos-hardware. The module itself can remain here. If you post the PR url, I will have a look.

Reply:
Depending on your motivation you can also add support here: https://github.com/Mic92/nixos-aarch64-images
This allows people with aarch64 to remaster our official aarch64 image with the custom u-boot.

Reply:
cc @samueldr

Reply:
As a complete reversal to my opinions about the kernel, U-Boot is "self-sustaining", since it's part of U-Boot mainline. Adding a build for a particular device is basically free.

We might want to stray away from including extraConfig. Sure, that means it'll have the default 3s boot delay, but that's how we build our U-Boots for now: stock as mainline does.

It's also quite cheap to build a custom U-Boot if needed by the end-user. That means it's not a huge deal if they need to customize it.

Except for: see my other comments about the patches.

Reply:
I'm not sure we want to introduce a non-mainline U-Boot into Nixpkgs. [I went as far as making myself honest](https://github.com/NixOS/nixpkgs/pull/98607), by removing the last non-mainline U-Boot from Nixpkgs.

A bit like for kernels, the main reasoning behind isn't about configuration or device-specificity, but about *maintainership*. Adding U-Boot defconfigs for a mainline board is trivial, and pretty much free.

Reply:
FWIW the U-Boot maintainer knows about these patches and wants this board fixed in U-Boot upstream.

Reply:
> Depending on your motivation you can also add support here: https://github.com/Mic92/nixos-aarch64-images This allows people with aarch64 to remaster our official aarch64 image with the custom u-boot.

Ok, I can do that when I have a working SD image.

Reply:
@ehmry you basically only need to know the offsets where the chip expects uboot to be: https://github.com/Mic92/nixos-aarch64-images/blob/main/images/rockchip.nix#L19

Reply:
@Mic92, be careful, that's not necessarily a given that an SoC will boot like Allwinners and Rockchips, by looking at a particular offset on the storage.

Though, here it looks like it may be, *when configured appropriately*.

 * https://community.nxp.com/t5/i-MX-Processors-Knowledge-Base/i-MX8-Boot-process-and-creating-a-bootable-image/ta-p/1101253